packages/mongodb-chatbot-server/src/routes/conversations/utils.ts

```diff
@@ -27,26 +27,54 @@ export const ApiConversation = z.object({
 
 export function convertMessageFromDbToApi(message: Message): ApiMessage {
   const { id, createdAt, role, content } = message;
-  const { rating, references } = message as Partial<AssistantMessage>;
-  return {
+  const apiMessage = {
     id: id.toString(),
     role,
     content,
     createdAt: createdAt.getTime(),
-    rating,
-    references,
   };
+  if(role === "assistant") {
+    const { rating, references } = message;
+    return {
+      ...apiMessage,
+      rating,
+      references,
+    };
+  }
+  return apiMessage;
+}
+
+function assertNever(x: never): never {
+  throw new Error("Unexpected object: " + x);
 }
+
+export function isMessageAllowedInApiResponse(message: Message) {
```

Comment:
```suggestion
function isMessageAllowedInApiResponse(message: Message) {
```
i don't think this is used externally